lib/helper/Playwright.js

```diff
@@ -0,0 +1,2328 @@
+ * * `waitForTimeout`: (optional) default wait* timeout in ms. Default: 1000.
+ * * `windowSize`: (optional) default window size. Set a dimension like `640x480`.
+ * * `userAgent`: (optional) user-agent string.
+ * * `manualStart`: (optional, default: false) - do not start browser before a test, start it manually inside a helper with `this.helpers["Playwright"]._startBrowser()`.
+ * * `browser`: (optional, default: chrome) - can be changed to `firefox` when using [Playwright-firefox](https://codecept.io/helpers/Playwright-firefox).
+ * * `chrome`: (optional) pass additional [Playwright run options](https://github.com/GoogleChrome/Playwright/blob/master/docs/api.md#Playwrightlaunchoptions).
+ *
+ *
+ * #### Example #1: Wait for 0 network connections.
+ *
+ * ```js
+ * {
+ *    helpers: {
+ *      Playwright : {
+ *        url: "http://localhost",
+ *        restart: false,
+ *        waitForNavigation: "networkidle0",
+ *        waitForAction: 500
+ *      }
+ *    }
+ * }
+ * ```
+ *
+ * #### Example #2: Wait for DOMContentLoaded event and 0 network connections
+ *
+ * ```js
+ * {
+ *    helpers: {
+ *      Playwright : {
+ *        url: "http://localhost",
+ *        restart: false,
+ *        waitForNavigation: [ "domcontentloaded", "networkidle0" ],
+ *        waitForAction: 500
+ *      }
+ *    }
+ * }
+ * ```
+ *
+ * #### Example #3: Debug in window mode
+ *
+ * ```js
+ * {
+ *    helpers: {
+ *      Playwright : {
+ *        url: "http://localhost",
+ *        show: true
+ *      }
+ *    }
+ * }
+ * ```
+ *
+ * #### Example #4: Connect to remote browser by specifying [websocket endpoint](https://chromedevtools.github.io/devtools-protocol/#how-do-i-access-the-browser-target)
+ *
+ * ```js
+ * {
+ *    helpers: {
+ *      Playwright: {
+ *        url: "http://localhost",
+ *        chrome: {
+ *          browserWSEndpoint: "ws://localhost:9222/devtools/browser/c5aa6160-b5bc-4d53-bb49-6ecb36cd2e0a"
+ *        }
+ *      }
+ *    }
+ * }
+ * ```
+ *
+* #### Example #5: Target URL with provided basic authentication
+ *
+ * ```js
+ * {
+ *    helpers: {
+ *      Playwright : {
+ *        url: 'http://localhost',
+ *        basicAuth: {username: 'username', password: 'password'},
+ *        show: true
+ *      }
+ *    }
+ * }
+ * ```
+ *
+ *
+ *
+ * Note: When connecting to remote browser `show` and specific `chrome` options (e.g. `headless` or `devtools`) are ignored.
+ *
+ * ## Access From Helpers
+ *
+ * Receive Playwright client from a custom helper by accessing `browser` for the Browser object or `page` for the current Page object:
+ *
+ * ```js
+ * const { browser } = this.helpers.Playwright;
+ * await browser.pages(); // List of pages in the browser
+ *
+ * const { page } = this.helpers.Playwright;
+ * await page.url(); // Get the url of the current page
+ * ```
+ *
+ * ## Methods
+ */
+class Playwright extends Helper {
+  constructor(config) {
+    super(config);
+
+    playwright = require('playwright');
```

Comment:
Maybe this variable must be the class property?

Reply:
not sure about it, it is global just the same as you do with `require('playwright')`.